From daf48c0cd937d52473f0bd9fb52b43e85a6e19ed Mon Sep 17 00:00:00 2001
Message-Id: <daf48c0cd937d52473f0bd9fb52b43e85a6e19ed.1429239283.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 31 Mar 2015 21:21:07 +0530
Subject: [PATCH 1/2] FOR_UPSTREAM [VPG]: drm/i915: Restore RC_PWRCTX_MAXCNT
 ring registers after per-engine reset

After per-engine reset current code in GFX driver does not re-initialize
rc6 related ring registers. This patch re-initialize RC_PWRCTX_MAXCNT to
expected value for better rc6 power saving.

Issue: GMINL-7576
Change-Id: I8164d9d09d0b0b26b86d9effe71bcdf7420b68a2
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 926cae3..7b1f97a 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1109,6 +1109,8 @@ int i915_handle_hung_ring(struct drm_device *dev, uint32_t ringid)
 		}
 	}
 
+	/* Restore RING_MAX_IDLE register post reset */
+	I915_WRITE(RING_MAX_IDLE(ring->mmio_base), 10);
 handle_hung_ring_error:
 	if (i915.enable_execlists)
 		i915_gem_context_unreference(current_context);
-- 
1.7.9.5

